### PR TITLE
fix for CVE-2023-0464 and CVE-2023-0465

### LINKED ISSUE #122: CVE for openssl 3.1.0
### Comment:

The following CVE's affects v3.1.0 openssl: https://nvd.nist.gov/vuln/detail/CVE-2023-0464, https://nvd.nist.gov/vuln/detail/CVE-2023-0465, https://nvd.nist.gov/vuln/detail/CVE-2023-0466. It seems that there are already patched and the new tag 3.1.0is available that is not affected by CVE's. 

When does conda-forge plan to release a new update?

### PR DESCRIPTION
**Pull request by ekomarova**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

Fixes #122

### CODE REVIEW

**Comment by ekomarova** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -12,9 +12,11 @@ source:
     - 0001-OBJ_nid2obj-Return-UNDEF-object-instead-of-NULL.patch
     # backport openssl/openssl#20613 as regression fix; can be dropped for v>3.1.0
     - 0002-Fix-some-patches-related-to-error-exiting.patch
+    - CVE-2023-0464.patch
+    - CVE-2023-0465.patch
```
Comment (ocefpaf) [inline recipe/meta.yaml:16]: Can you add links to the origin of those patches upstream?
That way we can track them and know when to drop them.

**Comment by h-vetinari** [comment]
Thanks for the PR!

I started to review, but got to the point where it was quicker that I fix things myself. For future reference:
* You were referencing the wrong PRs (need the ones for the branch corresponding to the version we're on)
* We should include all the relevant commits (especially relevant for testing that the backports work correctly)
* We should keep the commits message in the file name as generated by `git format-patch`
* Patches should go into `patches/` folder
* PRs don't need to be referenced twice, the short-form `openssl/openssl#xyz` is enough (GH will even turn it into a link)
* Since #122 was about tackling CVE-2023-0466 as well, I added a note why we're not backporting the relevant patch.

**Comment by h-vetinari** [comment]
@conda-forge/core 
The last available OpenSSL version(s) have some outstanding CVE's that OpenSSL itself [determined](https://www.openssl.org/news/secadv/20230322.txt) to be of low enough severity to not warrant a release immediately (3.1.1 is planned for May apparently).

Since CVE's show up on lots of people's radars, people are asking us to backport the CVE fixes. I'm personally fine with this (I double-checked; tests are included and passing), but would like to have your thoughts.

Assuming there are no objections (e.g. that we shouldn't be doing this for some reason), I'm planning to merge this in 48h.

**Comment by beckermr** [comment]
I really don't like carrying patches on security-sensitive items. I know it is a reality, but I think it is OK to say that if openssl determined it is OK to not make a release, we can trust them and not do it either. 

I personally do not want to be responsible for patches of this kind any more than I have to be.

**Comment by beckermr** [comment]
That all being said if others feel they have the knowhow and confidence that the patches are applied correctly, don't need further patches from upstream due to possible bugs in the patches themselves, and want to be responsible for them, I won't stand in their way.

**Comment by h-vetinari** [comment]
> I personally do not want to be responsible for patches of this kind any more than I have to be.

Haha, I _completely_ get that. Same here. 

> if others feel they have the knowhow and confidence that the patches are applied correctly, don't need further patches from upstream due to possible bugs in the patches themselves, and want to be responsible for them, I won't stand in their way.

In this case, the patches apply cleanly and come with tests (and for already comparatively low severity issues), don't have linked follow-ups on openssl tracker, so I'd be willing to put my name on that effort (barely).

**Comment by pkgw** [comment]
Thanks for putting in the work here! Everything that's been written so far seems very reasonable to me. Seeing as the "due diligence" has been put in already, I feel OK about merging this.

**Comment by ekomarova** [comment]
Thank you all for helping and considering this PR!

**Pull request merged**